src/app/categories/state/entry-categories.service.ts

```diff
@@ -42,14 +42,14 @@ export class EntryCategoriesService {
   /**
    * Updates the list of categories for the specified entry.
    */
-  updateEntryCategories(entryId: number) {
+  updateEntryCategories(entryId: number, published: boolean) {
     this.entryCategoriesStore.setLoading(true);
     this.entryCategoriesStore.setError(false);
     this.entryCategoriesStore.remove();
     this.currentSubscription?.unsubscribe();
-    this.currentSubscription = this.entriesService
-      .entryCategories(entryId)
-      .subscribe(
+    this.currentSubscription = null;
+    if (published) {
```

Comment:
This comment is for line 45/46 -- you should move that `...setLoading(true)` inside this if statement -- loading is never going to get set back to false if `published` is false.

Reply:
Good catch.  We should probably also `setLoading(false)` and `setError(false)` around where we unsubscribe, before the `if`.

Reply:
Basically, one way to think about `updateEntryCategories` is that it operates in two stages.  The first stage "cancels" the existing request and resets the store state, and the next stage fires off a new request, if necessary, that will set the store state when it completes...